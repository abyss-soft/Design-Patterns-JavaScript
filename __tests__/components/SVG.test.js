import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import SVG from '../../src/components/Svg';

const mockStore = configureMockStore();

describe('<SVG /> component', () => {
  it('renders JS control', () => {
    const store = mockStore({
      mode: 'dark'
    });

    const result = shallow(
      <div key="unique">
        <Provider store={store}>
          <SVG control="js" />
        </Provider>
      </div>
    );
    const tree = renderer.create(result).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders MODE control in DARK mode ', () => {
    const store = mockStore({
      mode: 'dark'
    });

    const result = shallow(
      <div key="unique">
        <Provider store={store}>
          <SVG control="mode" />
        </Provider>
      </div>
    );
    const tree = renderer.create(result).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders MODE control in LIGHT mode ', () => {
    const store = mockStore({
      mode: 'light'
    });

    const result = shallow(
      <div key="unique">
        <Provider store={store}>
          <SVG control="mode" />
        </Provider>
      </div>
    );
    const tree = renderer.create(result).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
