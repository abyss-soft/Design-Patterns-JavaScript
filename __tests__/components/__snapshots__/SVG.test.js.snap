// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVG /> component renders JS control 1`] = `
<div>
  <svg
    height="100%"
    viewBox="0 0 24 23"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <g
      transform="translate(0.000000,400.000000) scale(0.100000,-0.100000)"
    >
      <path
        d="M105,3929.3c-7.3-1.7-14.5-7.3-17.8-13.8c-1.9-3.8-2-4.7-2-12.2c0-7.3,0.2-8.5,1.8-12.1c3.8-8,10.3-13,26-19.7 c5.3-2.3,9.8-4.8,11.6-6.3c2.8-2.5,2.9-2.8,2.9-6.5c0-3.4-0.3-4.2-2-6.1c-2.6-2.7-5.8-3.7-11.4-3.7c-7.1,0.1-12.5,3-17.3,9.4 l-1.8,2.5l-7.8-4.5c-4.2-2.5-8-4.6-8.4-4.7c-1.1-0.4,3.8-7.3,7.8-11.1c11.4-10.3,32.9-13.1,46.8-6c11.7,5.9,17.2,18.8,13.8,32.1 c-2.8,10.6-9.3,16.4-27.6,24.3c-12.9,5.6-16,8.7-14.9,14.7c1.6,8.3,14.4,9,19.8,1.1c1.1-1.6,2.2-2.7,2.5-2.6c0.2,0.1,3.8,2.4,8,5.1 l7.5,4.8l-1.3,2.1c-2.1,3.3-7.2,8.2-10.3,10.1c-1.6,0.9-4.8,2.2-7.1,2.7C119.4,3930.2,109.3,3930.4,105,3929.3z"
      />
      <path
        d="M5,3880.7v-48.6h33.2h33.2v8.8v8.8H48.4H25.5v11.6v11.6h19.4h19.5l-0.2,8.7l-0.2,8.6l-19.2,0.2l-19.3,0.1v10.2v10.2h21.9 h21.9v9.2v9.2H37.1H5V3880.7z"
      />
      <path
        d="M181.5,3910.2c-7.9-10.6-15.6-21.3-17.1-23.8c-5.3-8.7-7.4-16.7-6.8-25.7c0.9-13.9,11.9-25.7,27.7-29.7 c5.4-1.4,17.1-1.3,22.6,0c6.8,1.8,12.8,4.9,17.2,9.2c6.8,6.4,9.9,14.2,9.9,24c0,12.7-6.3,22.8-17.4,28.1c-4.2,1.9-5.6,2.2-12.6,2.5 c-4.9,0.2-8.8,0-10.2-0.5c-2.1-0.7-2.1-0.6,1.1,2.1c2.5,2.3,22.9,28,25.4,32.2c0.4,0.6-2.3,0.8-12.4,0.8h-13L181.5,3910.2z M204,3876.5c5.6-2.9,8.7-9.9,7.1-16c-1.8-7-7.3-11.1-15-11.1c-11.3-0.1-18.6,11.4-13.5,21.1c2,3.8,3.7,5.3,8.1,6.8 C195.1,3878.9,199.9,3878.6,204,3876.5z"
      />
    </g>
  </svg>
</div>
`;

exports[`<SVG /> component renders MODE control in DARK mode  1`] = `
<div>
  <svg
    height="100%"
    viewBox="0 0 24 23"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79zM1 10.5h3v2H1zM11 .55h2V3.5h-2zm8.04 2.495l1.408 1.407-1.79 1.79-1.407-1.408zm-1.8 15.115l1.79 1.8 1.41-1.41-1.8-1.79zM20 10.5h3v2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm0 10c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4zm-1 4h2v2.95h-2zm-7.45-.96l1.41 1.41 1.79-1.8-1.41-1.41z"
    />
  </svg>
</div>
`;

exports[`<SVG /> component renders MODE control in LIGHT mode  1`] = `
<div>
  <svg
    height="100%"
    viewBox="0 0 24 23"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
    />
  </svg>
</div>
`;
